src/main/java/redis/clients/jedis/providers/MultiClusterPooledConnectionProvider.java

```diff
@@ -0,0 +1,321 @@
+                    return;
+
+            if (multiClusterIndex < 1 || multiClusterIndex > multiClusterMap.size())
+                throw new JedisValidationException("MultiClusterIndex: " + multiClusterIndex + " is not within " +
+                          "the configured range. Please choose an index between 1 and " + multiClusterMap.size());
+
+            validateTargetConnection(multiClusterIndex);
+
+            String originalClusterName = getClusterCircuitBreaker().getName();
+
+            if (activeMultiClusterIndex == multiClusterIndex)
+                log.warn("Cluster/database endpoint '{}' successfully closed its circuit breaker", originalClusterName);
+            else
+                log.warn("Cluster/database endpoint successfully updated from '{}' to '{}'",
+                         originalClusterName, getClusterCircuitBreaker(multiClusterIndex).getName());
+
+            activeMultiClusterIndex = multiClusterIndex;
+            lastClusterCircuitBreakerForcedOpen = false;
+        }
+    }
+
+    @Override
+    public void close() {
+        multiClusterMap.get(activeMultiClusterIndex).getConnectionPool().close();
+    }
+
+    @Override
+    public Connection getConnection() {
+        return multiClusterMap.get(activeMultiClusterIndex).getConnection();
+    }
+
+    public Connection getConnection(int multiClusterIndex) {
+        return multiClusterMap.get(multiClusterIndex).getConnection();
+    }
+
+    @Override
+    public Connection getConnection(CommandArguments args) {
+        return multiClusterMap.get(activeMultiClusterIndex).getConnection();
+    }
+
+    @Override
+    public Map<?, Pool<Connection>> getConnectionMap() {
+        ConnectionPool connectionPool = multiClusterMap.get(activeMultiClusterIndex).getConnectionPool();
+        return Collections.singletonMap(connectionPool.getFactory(), connectionPool);
+    }
+
+    public Cluster getCluster() {
+        return multiClusterMap.get(activeMultiClusterIndex);
+    }
+
+    public CircuitBreaker getClusterCircuitBreaker() {
+        return multiClusterMap.get(activeMultiClusterIndex).getCircuitBreaker();
+    }
+
+    public CircuitBreaker getClusterCircuitBreaker(int multiClusterIndex) {
+        return multiClusterMap.get(multiClusterIndex).getCircuitBreaker();
+    }
+
+    public boolean isLastClusterCircuitBreakerForcedOpen() {
+        return lastClusterCircuitBreakerForcedOpen;
+    }
+
+    public void runClusterFailoverPostProcessor(Integer multiClusterIndex) {
+        if (clusterFailoverPostProcessor != null)
+            clusterFailoverPostProcessor.accept(getClusterCircuitBreaker(multiClusterIndex).getName());
+    }
+
+    public void setClusterFailoverPostProcessor(Consumer<String> clusterFailoverPostProcessor) {
+        this.clusterFailoverPostProcessor = clusterFailoverPostProcessor;
+    }
+
+    public static class Cluster {
```

Comment:
In our documentation we are saying _Redis deployments_. Even more so this PR is done in a way this supports standalone Redis instances. So each of the deployments wouldn't necessarily be a _Cluster_ deployment. (Am I correct so far?)

This inner `Cluster` class is actually represents a single deployment, right? If so, the word _Cluster_ is not appropriate name for this class.

@aterleto Please this class name.

Reply:
May I suggest a set of renamings?
```
            MultiClusterJedisClientConfig => MultiDeploymentConfig / MultiDeploymentClientConfig
(ic)             ClusterJedisClientConfig => DeploymentConfig / SingleDeploymentConfig
     MultiClusterPooledConnectionProvider => MultiDeploymentConnectionProvider
(ic)                              Cluster => Deployment / SingleDeployment
```
(ic = inner class)